### PR TITLE
On SCA mode, stop throwing an error during dry-run auto-attach

### PR DESCRIPTION
**Pull request by sonalidhome**
- Added more spec tests around SCA mode.

### CODE REVIEW

**Review by Ceiu — CHANGES_REQUESTED** [review]
This doesn't really move the bar substantially as far as testing in hosted vs standalone modes. What I would expect here is to make the "before(:each)" block be hosted-aware and change how/where it creates the necessary data accordingly.

For individual tests which simply do not work because they require an endpoint that is disallowed in hosted mode, those tests should then be individually skipped based on the mode. However, given the current changes toward SCA mode everywhere, I would expect every operation to be testable in some capacity regardless of mode.

**Comment by sonalidhome** [comment]
Updated the PR. Added a few more tests around SCA mode in hosted.

File: server/spec/content_access_spec.rb [hunk 1]
```diff
@@ -883,4 +883,25 @@ def regenerate_cert_test(&updater)
...
+    # System purpose status
+    status = @cp.get_purpose_compliance(consumer['uuid'])
+    status['status'].should == 'disabled'
+
+    # compliance status
+    status = @cp.get_compliance(consumer['uuid'])
+    status['status'].should == 'disabled'
+  end
+
+  it "should not allow auto-attach for consumer with activation keys" do
```
Comment (nikosmoum) [inline server/spec/content_access_spec.rb:899]: This test is wrong, so let's remove it. The scenario it describes is opposite of what we want, meaning registration should _never_ fail in SCA mode, regardless of if activation keys are used, or if they have auto-attach set to true or not.
The reason the test passes is unrelated: it seems to be that you accidentally set the value 'true' as the owner key on the register() method.

File: server/spec/content_access_spec.rb [hunk 2]
```diff
@@ -883,4 +883,25 @@ def regenerate_cert_test(&updater)
     end
   end
 
+  it 'should be disabled for owner in SCA mode' do
```
Comment (nikosmoum) [inline server/spec/content_access_spec.rb:886]: This test was missing, so let's keep it :+1:

File: server/spec/consumer_resource_spec.rb [hunk 3]
```diff
@@ -1438,6 +1438,12 @@ def verify_installed_pids(consumer, product_ids)
     expected_exceptions.each { |e| e.should be_an(RestClient::Gone) | be_an(RestClient::ResourceNotFound) }
   end
 
+  it "should not allow to content access certificate body when not SCA mode" do
```
Comment (nikosmoum) [inline server/spec/consumer_resource_spec.rb:1441]: :+1:

File: server/spec/autobind_disabled_for_owner_spec.rb [hunk 4]
```diff
@@ -154,4 +154,26 @@
     job['state'].should == "FAILED"
     job['resultData'].should == "org.candlepin.async.JobExecutionException: Auto-attach is disabled for owner #{owner['key']} because of the content access mode setting."
   end
+
+  it "should not dryrun when owner is in SCA mode" do
```
Comment (nikosmoum) [inline server/spec/autobind_disabled_for_owner_spec.rb:158]: The premise of this test is also wrong, and it actually exposes a gap we missed.
When we did the changes to no-op  auto-attach and return status 200 when in SCA mode (https://github.com/candlepin/candlepin/pull/2947), we should have _also_ applied that change to the dry-run auto-attach. Let's apply that change now, and change this test to assert the opposite: that dry-run-auto-attach returns 200 when on SCA mode

**Review by nikosmoum — CHANGES_REQUESTED** [review]
1) Let's not make SCA mode testable in standalone mode in this commit, since we're going to change that anyway. We'll do the necessary changes as part of that change.
2) A couple of changes are needed.
3) Let's change the commit message to reflect the behavioural change on dry-run auto-attach instead since we're no longer doing what the original task was. Something like this should do:
```
On SCA mode, stop throwing an error during dry-run auto-attach

- Added more spec tests around SCA mode
```

File: server/spec/content_access_spec.rb (hunk 1 above)
Comment (sonalidhome) [reply server/spec/content_access_spec.rb]: Done.

File: server/spec/autobind_disabled_for_owner_spec.rb (hunk 4 above)
Comment (sonalidhome) [reply server/spec/autobind_disabled_for_owner_spec.rb]: Updated.

File: server/spec/content_access_spec.rb [hunk 5]
```diff
@@ -883,4 +883,25 @@ def regenerate_cert_test(&updater)
     end
   end
```
[the remark this answers is not in the document]
Comment (sonalidhome) [reply server/spec/content_access_spec.rb:884]: ack.

**Pull request merged**